exchanges/okgroup/okgroup_websocket.go

```diff
@@ -292,113 +233,174 @@ func (o *OKGroup) WsLogin() error {
+		if o.Verbose {
+			log.Debugf(log.ExchangeSys,
+				"WS Error Event: %v Message: %v for %s",
+				errorResponse.Event,
+				errorResponse.Message,
+				o.Name)
+		}
 
-	return channel[:index]
-}
-
-// GetAssetTypeFromTableName gets the asset type from the table name
-// eg "spot/ticker:BTCUSD" results in "SPOT"
-func (o *OKGroup) GetAssetTypeFromTableName(table string) asset.Item {
-	assetIndex := strings.Index(table, "/")
-	switch table[:assetIndex] {
-	case asset.Futures.String():
-		return asset.Futures
-	case asset.Spot.String():
-		return asset.Spot
-	case "swap":
-		return asset.PerpetualSwap
-	case asset.Index.String():
-		return asset.Index
-	default:
-		log.Warnf(log.ExchangeSys, "%s unhandled asset type %s",
+		return fmt.Errorf("%v error - %v message: %s ",
 			o.Name,
-			table[:assetIndex])
-		return asset.Item(table[:assetIndex])
+			errorResponse.ErrorCode,
+			errorResponse.Message)
+	}
+	var eventResponse WebsocketEventResponse
+	err = json.Unmarshal(respRaw, &eventResponse)
+	if err == nil && eventResponse.Event != "" {
+		if eventResponse.Event == "login" {
+			o.Websocket.SetCanUseAuthenticatedEndpoints(eventResponse.Success)
+		}
+		if o.Verbose {
+			log.Debugf(log.ExchangeSys,
+				"WS Event: %v on Channel: %v for %s",
+				eventResponse.Event,
+				eventResponse.Channel,
+				o.Name)
+		}
 	}
+	return nil
 }
 
-// WsHandleDataResponse classifies the WS response and sends to appropriate handler
-func (o *OKGroup) WsHandleDataResponse(response *WebsocketDataResponse) {
-	switch o.GetWsChannelWithoutOrderType(response.Table) {
-	case okGroupWsCandle60s, okGroupWsCandle180s, okGroupWsCandle300s,
-		okGroupWsCandle900s, okGroupWsCandle1800s, okGroupWsCandle3600s,
-		okGroupWsCandle7200s, okGroupWsCandle14400s, okGroupWsCandle21600s,
-		okGroupWsCandle43200s, okGroupWsCandle86400s, okGroupWsCandle604900s:
-		o.wsProcessCandles(response)
-	case okGroupWsDepth, okGroupWsDepth5:
-		// Locking, orderbooks cannot be processed out of order
-		orderbookMutex.Lock()
-		err := o.WsProcessOrderBook(response)
-		if err != nil {
-			for i := range response.Data {
-				a := o.GetAssetTypeFromTableName(response.Table)
-				var c currency.Pair
-				switch a {
-				case asset.Futures, asset.PerpetualSwap:
-					f := strings.Split(response.Data[i].InstrumentID, delimiterDash)
-					c = currency.NewPairWithDelimiter(f[0]+delimiterDash+f[1], f[2], delimiterDash)
-				default:
-					f := strings.Split(response.Data[i].InstrumentID, delimiterDash)
-					c = currency.NewPairWithDelimiter(f[0], f[1], delimiterDash)
-				}
-
-				channelToResubscribe := wshandler.WebsocketChannelSubscription{
-					Channel:  response.Table,
-					Currency: c,
-				}
-				o.Websocket.ResubscribeToChannel(channelToResubscribe)
-			}
-		}
-		orderbookMutex.Unlock()
-	case okGroupWsTicker:
-		o.wsProcessTickers(response)
-	case okGroupWsTrade:
-		o.wsProcessTrades(response)
+func stringToOrderStatus(num string) order.Status {
```

Comment:
After reviewing a lot of these in this PR and considering the nature of this, it would be good to add test coverage for these conversion funcs